Plugins/NativeEngine/Source/NativeEngine.cpp

```diff
@@ -721,13 +709,54 @@ namespace Babylon
             }};
 
         auto vertexShader = bgfx::createShader(bgfx::copy(shaderInfo.VertexBytes.data(), static_cast<uint32_t>(shaderInfo.VertexBytes.size())));
-        InitUniformInfos(vertexShader, shaderInfo.UniformStages, program->UniformInfos, program->UniformNameToIndex);
-        program->VertexAttributeLocations = std::move(shaderInfo.VertexAttributeLocations);
+        InitUniformInfos(vertexShader, shaderInfo.UniformStages, program.UniformInfos, program.UniformNameToIndex);
+        program.VertexAttributeLocations = std::move(shaderInfo.VertexAttributeLocations);
```

Comment:
To do this properly, ProgramData should not be passed in since we only want ProgramData to be accessed from the JS thread. This function should return both the bgfx::ProgramHandle and a VertexAttributeLocations map.

Reply:
I've changed this, the test seems to work but please let me know if there's something wrong with the new code.

Reply:
Sorry, the same issue for UniformInfos and UniformNameToIndex. Basically, we want to avoid accessing things in ProgramData unless we own ProgramData. Another option is to make ProgramData movable so that CreateProgramInternal returns a ProgramData and then the outer function can move the ProgramData into the pointer version.

This is more for cleanliness than correctness. It should also address the other issue.

Reply:
I pushed the changes we discussed (making `ProgramData` movable) and hopefully that addresses this issue.